src/main/scala/org/broadinstitute/clio/dataaccess/CachedServerStatusDAO.scala

```diff
@@ -0,0 +1,47 @@
+package org.broadinstitute.clio.dataaccess
+
+import org.broadinstitute.clio.model.ServerStatusInfo
+
+import scala.concurrent.Future
+import scala.util.Try
+
+/**
+  * Stores the server status in a synchronized mutable variable.
+  */
+class CachedServerStatusDAO private() extends ServerStatusDAO {
+  private val mutex = new Object()
+  private var currentStatus: ServerStatusInfo = ServerStatusInfo.NotStarted
+
+  override def setStatus(status: ServerStatusInfo): Future[Unit] = Future.fromTry(
+    Try {
+      mutex synchronized {
+        import CachedServerStatusDAO.level
+        if (level(currentStatus) < level(status))
+          currentStatus = status
+      }
+    }
+  )
+
+  override def getStatus: Future[ServerStatusInfo] = Future.fromTry(
+    Try {
+      mutex synchronized currentStatus
+    }
+  )
+}
+
+object CachedServerStatusDAO {
+  def apply(): CachedServerStatusDAO = {
+    new CachedServerStatusDAO()
+  }
+
+  private def level(status: ServerStatusInfo): Int = {
+    status match {
+      case ServerStatusInfo.NotStarted => 0
```

Comment:
Would it be possible to store this information directly in `ServerStatusInfo`, e.g. as per-instance state? Doing that avoids the need to handle the "unknown" case and prevents the situation where a user adds a new status and forgets to add it here.